### PR TITLE
Fix cppcheck uninitMemberVar warnings in src/core/

### PR DESCRIPTION
**Pull request by rouault**
Found with
~/cppcheck/cppcheck --enable=all --inconclusive ../src/core --library=$HOME/cppcheck/cfg/qt.cfg -j 8 -DSIP_OUT 2>&1 | grep uninitMemberVar

### CODE REVIEW

File: src/core/qgsfeaturerequest.cpp [hunk 1]
```diff
@@ -56,7 +56,7 @@ QgsFeatureRequest::QgsFeatureRequest( const QgsExpression &expr, const QgsExpres
 {
 }
 
-QgsFeatureRequest::QgsFeatureRequest( const QgsFeatureRequest &rh )
+QgsFeatureRequest::QgsFeatureRequest( const QgsFeatureRequest &rh ): mTransformContext{}
```
Comment (rouault) [inline src/core/qgsfeaturerequest.cpp]: I'm wondering here if mTransformContext shouldn't rather be copied by the operator= ?

Comment (nyalldawson) [reply src/core/qgsfeaturerequest.cpp]: Yes, you're right...

File: src/core/effects/qgsgloweffect.cpp [hunk 2]
```diff
@@ -33,6 +33,8 @@ QgsGlowEffect::QgsGlowEffect( const QgsGlowEffect &other )
   , mSpreadUnit( other.spreadUnit() )
   , mSpreadMapUnitScale( other.spreadMapUnitScale() )
   , mBlurLevel( other.blurLevel() )
+  , mBlurUnit( other.mBlurUnit )
+  , mBlurMapUnitScale( other.mBlurMapUnitScale )
```
Comment (nyalldawson) [inline src/core/effects/qgsgloweffect.cpp]: Likely fixes https://github.com/qgis/QGIS/issues/34089

File: src/core/qgsfeaturerequest.cpp (hunk 1 above)
Comment (rouault) [reply src/core/qgsfeaturerequest.cpp]: done

File: src/core/effects/qgsgloweffect.cpp (hunk 2 above)
Comment (rouault) [reply src/core/effects/qgsgloweffect.cpp]: No, it doesn't

Comment (nyalldawson) [reply src/core/effects/qgsgloweffect.cpp]: Maybe because it's missing in the operator= method?

Comment (rouault) [reply src/core/effects/qgsgloweffect.cpp]: They were indeed missing in operator=, but fixing it doesn't fix the issue. Perhaps a pure GUI issue

Comment (nyalldawson) [reply src/core/effects/qgsgloweffect.cpp]: Yep - gui issue. I'll fix after this is merged.

Comment (rouault) [reply src/core/effects/qgsgloweffect.cpp]: > I'll fix after this is merged.

I'm on it. Fix under way

Comment (rouault) [reply src/core/effects/qgsgloweffect.cpp]: Fixed by https://github.com/qgis/QGIS/pull/34135/commits/01412a7bec32d736d13a6166f3ecbc2f5230284c . The fix is only effective if combined with the copy constructor/operator= fix . So this whole PR should probably be backported

**Pull request merged**